modules/SofaExporter/CMakeLists.txt

```diff
@@ -81,6 +81,13 @@ sofa_generate_package(
     )
 
 # Tests
+
+if (SOFA_BUILD_TESTS)
+    option(SOFAEXPORTER_BUILD_TESTS "Compile the automatic tests" ON)
+else()
+    set(SOFAEXPORTER_BUILD_TESTS FALSE)
+endif()
```

Comment:
I don't agree with this approach. 
SOFA_BUILD_TEST is for SOFA tests. SofaExporter is now modularized so it should be considered as out-of-SOFA.

My view is:
- SOFA old modules (that are not real modules) do not integrate their tests themselves because of SofaTest dependencies (see the order of `add_subdirectory` calls in main CMakeLists)
- new modules ("modularized") and plugins and projects must be buildable out of SOFA so they integrate their tests themselves and do not depend on a SOFA internal variable. Those tests, though, can have a SofaTest dependency (SofaExporter_test does) and thus may fail to generate if SofaTest is not found.

In this case for example, I prefer an explicit SOFAEXPORTER_BUILD_TESTS that may lead to a failure because SofaExporter_test couldn't find SofaTest than a magic check above SofaExporter_test setting the variable OFF beforehand.

What could be changed though is the way SofaExporter_test finds SofaTest
```
find_package(SofaTest QUIET)
if(NOT SofaTest_FOUND)
    message(SEND_ERROR "Couldn't find SofaTest. If building SOFA, you need to activate SOFA_BUILD_TESTS")
endif()
```

Reply:
ok I understand that tests from module should be independant but they need SofaTest lib.
In my case I don't compile SofaTest (bad I know).  But unchecking SofaTest failed at cmake configuration because SOFAEXPORTER_BUILD_TESTS need SofaTest. So it means I would need to uncheck each TEST in the cmake configuration...

Maybe the good approach is (not sure it works): 
```
if (SOFA_TEST)
    option(SOFAEXPORTER_BUILD_TESTS "Compile the automatic tests" ON)
else()
    option(SOFAEXPORTER_BUILD_TESTS "Compile the automatic tests" OFF)
endif()
```

and in the SOFAExporter_test cmake:

```
find_package(SofaTest REQUIRED)
if(NOT SofaTest_FOUND)
    message(SEND_ERROR "Couldn't find SofaTest. If building SOFA, you need to activate SOFA_BUILD_TESTS")
endif()
```